ecosystem/node-checker/vfn-check-client/src/check.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context;
+use anyhow::Result;
+use aptos_node_checker_lib::EvaluationSummary;
+use aptos_sdk::rest_client::Client as AptosClient;
+use aptos_sdk::types::account_address::AccountAddress;
+use aptos_sdk::types::account_config::CORE_CODE_ADDRESS;
+use aptos_sdk::types::network_address::NetworkAddress;
+use aptos_sdk::types::on_chain_config::ValidatorSet;
+use aptos_sdk::types::validator_info::ValidatorInfo;
+use clap::Parser;
+use log::info;
+use reqwest::Client as ReqwestClient;
+use reqwest::Url;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::SystemTime;
+use std::time::UNIX_EPOCH;
+use tokio::sync::Semaphore;
+
+use crate::helpers::extract_network_address;
+
+// Unfortunately we don't have any way, on chain or not, to know what the API
+// port is, so we just assume it is 80 or 8080. If their VFN is inaccessible at 80,
+// e.g. because it runs on 7777 and they have an LB in front of it that listens
+// at 80, we're just out of luck. If we get this kind of information from
+// elsewhere at some point, we could look at joining it here.
+pub const API_PORTS: &[u16] = &[80, 8080];
```

Comment:
Annoying... In a non code-freeze world, we could have exposed this using the REST API 😄

Reply:
Alas, we don't know what the REST API is, it's not on chain nor given to us any other way :p